fluentd/fluent.conf

```diff
@@ -225,7 +225,9 @@
     type record_transformer
     enable_ruby
     <record>
-      hostname ${host}
+      # if host == 'localhost' do a fqdn lookup
+      ## we pull the hostname from the host's /etc/hostname mounted at /etc/docker-hostname to correctly identify the hostname
+      hostname ${host.eql?('localhost') ? (File.open('/etc/docker-hostname') { |f| f.readline }.rstrip || host) : host}
```

Comment:
the null check condition i'm using here doesn't actually do what i was hoping it would do, which is to use host if we cant get a value from /etc/docker-hostname for some reason.